docs/_documentations/debugging.md

```diff
@@ -10,10 +10,33 @@ type: document
 
 # Debugging in Codewind
 
-Complete the following steps to enable debugging in Codewind applications:
+Complete the following steps to debug in Codewind applications:
 
 1. In the Codewind view, right-click a project and select **Restart in Debug Mode** from the menu.
 2. Wait for the project state to switch to **Debugging**. After it switches to Debugging mode, the debugger is attached.
 3. When in Debugging mode and when the debugger is attached, you can debug your application. Perform debug actions that are supported by the IDE that you are using.
 4. While the project is still in Debugging mode, if the debug process is detached, right-click your project and select **Attach Debugger** from the menu. This action reattaches the debugger to the debug process, and you can continue with debugging your application.
-5. To switch back to run mode, right-click your project and select **Restart in Run Mode** from the menu.
\ No newline at end of file
+5. To switch back to run mode, right-click your project and select **Restart in Run Mode** from the menu.
+
+## Using Java hot code replace while debugging
+If you want to use Java hot code replace and change your code while you debug, disable automatic builds.
+1. To disable automatic builds, right-click your project in the Codewind Explorer view and select **Disable Auto Build**.
+2. If you want to start a build while automatic builds are disabled, right-click your project and select **Build**.
+3. Enable automatic builds again after you finish debugging. To enable automatic builds again, right-click your project and select **Enable Auto Build**.
+
+## Debugging initialization code for Java projects
+- If you want to debug initialization code, set breakpoints in the code. You can also set breakpoints in your application code.
+- When you debug initialization code, wait for the project state to change to **Debugging** or for the debugger to stop at a breakpoint.
+- Normally you can reload your application multiple times to isolate a problem. However, when you debug initialization code, restart your project in debug mode to stop in the code again.
+
+## Debugging Node.js projects
```

Comment:
Is this a shared document with VSCode and IntelliJ? This part does not apply to VSCode. Not sure about IntelliJ.

Reply:
John specified that it doesn't apply to IntelliJ. So looks like this Node.js section applies to Eclipse only. I will update the docs accordingly. :) Thanks!